docs: Update changelog.md references (#14046)
shopware/platform: CHANGELOG.md

# Changelog

> > ⚠️ **Note on changelog structure (Shopware 6.7.5 and newer)**  

Reviewer: ```suggestion
> [!WARNING]
> **Note on changelog structure (Shopware 6.7.5 and newer)**  
```

could help highlighting the disclaimer, see https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

Reviewer 2: nice i didn't knew that, makes sense to add 

however i would keep the warning icon as well, for cases where you read the raw content either on the web or in the IDE

Reviewer: looks kinda odd with the double warning, but another alert type (note, important) would also do it

<img width="501" height="90" alt="image" src="https://github.com/user-attachments/assets/99327410-b609-4475-ba71-21f471285fd4" />

>  

> > Starting with **Shopware 6.7.5**, this file is no longer the main source of truth for release changes.

Reviewer 2: ```suggestion
> Starting with **Shopware 6.7.5**, this file is no longer updated with the recent changes.
```

>  

> > - **Curated, developer-facing release information** can be found in the versioned release info files, e.g.:  

Reviewer 2: ```suggestion

> - **Curated, developer-facing release information** can be found in the versioned [release info files](./RELEASE_INFO-6.7.md).  
```

Let's make use of md linking feature, here having the relative link should also work when you read in IDE as well as when you read in the web

Author: Oh yes, thank you.

# Changelog

> [!WARNING]
> **Note on changelog structure (Shopware 6.7.5 and newer)**  
>  
> Starting with **Shopware 6.7.5**, this file is no longer updated with the recent changes.
>  

> 

Reviewer: ```suggestion
>
```

i guess 4278816fd5dfc493949105cb144f89c30b46fa1e broke the formatting of the rendered alert due to the extra line 

<img width="881" height="364" alt="Image" src="https://github.com/user-attachments/assets/716148e6-501d-4a10-9a3c-c412fa5d047c" />
